File: server/src/main/java/com/oceanbase/ocp/controller/PropertyController.java
/*
 * Copyright (c) 2023 OceanBase
 * OCP Express is licensed under Mulan PSL v2.
 * You can use this software according to the terms and conditions of the Mulan PSL v2.
 * You may obtain a copy of Mulan PSL v2 at:
 *          http://license.coscl.org.cn/MulanPSL2
 * THIS SOFTWARE IS PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES OF ANY KIND,
 * EITHER EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO NON-INFRINGEMENT,
 * MERCHANTABILITY OR FIT FOR A PARTICULAR PURPOSE.
 * See the Mulan PSL v2 for more details.
 */

package com.oceanbase.ocp.controller;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Pageable;
import org.springframework.data.domain.Sort;
import org.springframework.data.web.PageableDefault;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import com.oceanbase.ocp.core.property.PropertyService;
import com.oceanbase.ocp.core.property.SystemInfo;
import com.oceanbase.ocp.core.property.model.PropertyMeta;
import com.oceanbase.ocp.core.response.NoDataResponse;
import com.oceanbase.ocp.core.response.PaginatedResponse;
import com.oceanbase.ocp.core.response.ResponseBuilder;
import com.oceanbase.ocp.core.response.SuccessResponse;

import lombok.extern.slf4j.Slf4j;

@RestController
@Slf4j
@RequestMapping("/api/v1")
public class PropertyController {

    @Autowired
    private PropertyService propertyService;
    @Autowired
    private SystemInfo systemInfo;

    @GetMapping("/config/properties")
    public PaginatedResponse<PropertyMeta> findNonFatalProperties(
            @RequestParam(name = "keyLike", required = false) String keyLike,
            @PageableDefault(size = Integer.MAX_VALUE, sort = {"key"},
                    direction = Sort.Direction.ASC) Pageable pageable) {
        return ResponseBuilder.paginated(propertyService.findAllVisibleProperties(keyLike, pageable));
    }

    @PutMapping("/config/properties/{id:[\\d]+}")
    public SuccessResponse<PropertyMeta> updateProperty(@PathVariable long id, @RequestParam String newValue) {
        return ResponseBuilder.single(propertyService.updateProperty(id, newValue));
    }

    @PostMapping("/config/refresh")
    public NoDataResponse refreshProperties() {
        propertyService.refresh();
        return ResponseBuilder.noData();
    }

    @GetMapping("/config/systemInfo")
    public SuccessResponse<SystemInfo> getSystemInfo() {
        return ResponseBuilder.single(systemInfo);
    }

}

File: frontend/src/service/ocp-express/PropertyController.ts
/*
 * Copyright (c) 2023 OceanBase
 * OCP Express is licensed under Mulan PSL v2.
 * You can use this software according to the terms and conditions of the Mulan PSL v2.
 * You may obtain a copy of Mulan PSL v2 at:
 *          http://license.coscl.org.cn/MulanPSL2
 * THIS SOFTWARE IS PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES OF ANY KIND,
 * EITHER EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO NON-INFRINGEMENT,
 * MERCHANTABILITY OR FIT FOR A PARTICULAR PURPOSE.
 * See the Mulan PSL v2 for more details.
 */

/* eslint-disable */
// 该文件由 OneAPI 自动生成，请勿手动修改！
import request from '@/util/request';

/** 此处后端没有提供注释 GET /api/v1/config/properties */
export async function findNonFatalProperties(
  params: {
    // query
    keyLike?: string;
    pageable?: API.Pageable;
  },
  options?: { [key: string]: any },
) {
  return request<API.PaginatedResponse_PropertyMeta_>('/api/v1/config/properties', {
    method: 'GET',
    params: {
      ...params,
      pageable: undefined,
      ...params['pageable'],
    },
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 GET /api/v1/config/systemInfo */
export async function getSystemInfo(options?: { [key: string]: any }) {
  return request<API.SuccessResponse_SystemInfo_>('/api/v1/config/systemInfo', {
    method: 'GET',
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 POST /api/v1/config/refresh */
export async function refreshProperties(options?: { [key: string]: any }) {
  return request<API.NoDataResponse>('/api/v1/config/refresh', {
    method: 'POST',
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 PUT /api/v1/config/properties/${param0} */
export async function updateProperty(
  params: {
    // query
    newValue?: string;
    // path
    id?: number;
  },
  options?: { [key: string]: any },
) {
  const { id: param0, ...queryParams } = params;
  return request<API.SuccessResponse_PropertyMeta_>(`/api/v1/config/properties/${param0}`, {
    method: 'PUT',
    params: {
      ...queryParams,
    },
    ...(options || {}),
  });
}
